### PR TITLE
Network worker

### PR DESCRIPTION
**Pull request by jamesscottbrown**

### CODE REVIEW

File: workers/tulip/__init__.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+        for algorithm in algorithms:
+            default_params = tlp.getDefaultPluginParameters(algorithm)
+
+            for field in default_params:
+                if field != 'result':
+                    field_name = algorithm + "_" + field
+
+                    properties[field_name] = {
+                        'type': 'string',
```
Comment (oserban) [inline workers/tulip/__init__.py]: This can be a checkbox if the default parameter is bool

File: workers/tulip/__init__.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+    def process(self, project_name: str, meta: OveMeta, options: Dict):
+        logging.info("Copying %s/%s into the temp place ...", project_name, meta.name)
+
+        if not options:
+            options = {}
+
+        with TemporaryDirectory() as folder:
+
+            with TemporaryDirectory() as folder2:
```
Comment (oserban) [inline workers/tulip/__init__.py]: Please rename folder2 into some more useful (e.g. resource folder or asset folder)

File: workers/tulip/__init__.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        meta.index_file = os.path.join(meta.worker_root + self.name, base_name, result_name)
+
+        self._file_controller.set_asset_meta(project_name, meta.name, meta)
+        logging.info("Finished generating %s/%s into the storage ...", project_name, meta.name)
+
+    @staticmethod
+    def convert_param(param):
+        if not isinstance(param, str):
+            return param
+        elif str.lower(param) == 'true':
```
Comment (oserban) [inline workers/tulip/__init__.py]: for str to bool conversion there is a function called to_bool in the commons.util module

File: docker/worker-tulip/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,22 @@
+FROM python:3-slim
```
Comment (oserban) [inline docker/worker-tulip/Dockerfile]: Is there a particular reason for not using the pypy image?

Comment (jamesscottbrown) [reply docker/worker-tulip/Dockerfile]: Yes - the ability to `pip install tulip-python`

File: workers/tulip/__init__.py (hunk 3 above)
Comment (jamesscottbrown) [reply workers/tulip/__init__.py]: That function throws an exception if it passed something that is not a string representation of a bool: I just want to defensively ensure that *if* a string is a representation of a boolean then it is converted to an actual boolean.

Comment (oserban) [reply workers/tulip/__init__.py]: I realized that the to_bool function does not do the conversion you require

**Pull request merged**